walk-in-interview/src/main/webapp/common-functions.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * This file wraps some commonly used job related function as a reusable module.
+ */
+
+/**
+ * Gets the requirements list from the servlet
+ * (which gets it from the database).
+ * @return {Object} the requirements list in a key-value mapping format.
+ */
+function getRequirementsList() {
+    // TODO(issue/17): GET request to servlet to get from database
+    // returning some hardcoded values for now
+    return {
+      'O_LEVEL': 'O Level',
+      'LANGUAGE_ENGLISH': 'English',
+      'DRIVING_LICENSE_C': 'Category C Driving License',
+    };
+}
+
+/**
+ * Gets job detail from user input.
+ * @return {Object} containing the user inputs.
+ */
+function getJobDetailsFromUserInput() {
+    const name = document.getElementById('title').value;
+    const description = document.getElementById('description').value;
+    const address = document.getElementById('address').value;
+    const postalCode = document.getElementById('postal-code').value;
+    const payFrequency = document.getElementById('pay-frequency').value;
+    const payMin = document.getElementById('pay-min').valueAsNumber;
+    const payMax = document.getElementById('pay-max').valueAsNumber;
+
+    const requirementsCheckboxes =
+        document.getElementsByName('requirements-list');
+    const requirementsList = [];
+    requirementsCheckboxes.forEach(({checked, id}) => {
+        if (checked) {
+        requirementsList.push(id);
+        }
+    });
+
+    const expiry = document.getElementById('expiry').valueAsNumber;
+    const duration = document.getElementById('duration').value;
+
+    const jobDetails = {
+        jobTitle: name,
+        jobLocation: {
+        address: address,
+        postalCode: postalCode,
+        lat: 1.3039, // TODO(issue/13): get these from places api
+        lon: 103.8358,
+        },
+        jobDescription: description,
+        jobPay: {
+        paymentFrequency: payFrequency,
+        min: payMin,
+        max: payMax,
+        },
+        requirements: requirementsList,
+        postExpiryTimestamp: expiry,
+        jobDuration: duration,
+    };
+
+    return jobDetails;
+}
+
+/**
+ * Validates the user input
+ * @return {boolean} depending on whether the input is valid or not.
+ */
+function validateRequiredUserInput() {
+    // TODO(issue/19): add more validation checks
+    const name = document.getElementById('title');
+    const description = document.getElementById('description');
+    const address = document.getElementById('address');
+    const postalCode = document.getElementById('postal-code');
+    const payFrequency = document.getElementById('pay-frequency').value;
+    const payMin = document.getElementById('pay-min').valueAsNumber;
+    const payMax = document.getElementById('pay-max').valueAsNumber;
+    const duration = document.getElementById('duration').value;
+    const expiry = document.getElementById('expiry').valueAsNumber;
+
+    if (name.value === '') {
+        setErrorMessage(/* msg */ name.placeholder,
```

Comment:
you changed the setErrorMessage params. There should be a third param here for the `defaultMessage`. I would suggest that you leave the setErrorMessage function as it was before, and import `string.en.js` into this file.
```
function setErrorMessage(msg, includesDefault) {
  document.getElementById('error-message').innerText =
    (includesDefault ? STRINGS['error-message'] + msg : msg);
}
```

Reply:
or you could simply have a constant for `STRINGS['error-message']` instead of importing it. i.e. `const DEFAULT_ERROR_MSG = 'xxx';`